### PR TITLE
Add Admin Layout

### PR DESCRIPTION
**Pull request by Lordapa**

### CODE REVIEW

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+#main-layout {
+  height: 100%;
+}
+
+.admin-layout {
+  margin-top: 4.4em;
+}
+
+#main-layout .header {
+  padding: 0 20px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css:17]: units should be em or % or view port units

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+}
+#main-layout .header .logo {
+  width: auto;
+  height: 31px;
+  margin: 0px 24px 16px 0;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css:22]: units should be em or % or view port units

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+}
+
+.menu-link {
+  display: flex;
+  align-items: center;
+}
+
+svg {
+  margin-right: 6px;
+  font-size: 24px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css:35]: units should be em or % or view port units

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+.create-button {
+  float: right;
+}
+
+.section-title {
+  display: inline;
+}
+
+.ant-btn {
+  margin-right: 8px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.css:61]: units should be em or % or view port units

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.html [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<nz-layout id="main-layout">
+  <app-nav-menu></app-nav-menu>
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.component.html:2]: the app-nav-menu, should be in every component or in a parent layout? @Faculewis

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.module.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+const ICONS: IconDefinition[] = [HeartOutline, TeamOutline, ProfileOutline];
+
+@NgModule({
+  imports: [
+    NzIconModule.forChild(ICONS),
+    AdminRoutingModule,
+    CommonModule,
+    NzLayoutModule,
+    NzMenuModule,
+    TranslateModule,
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.module.ts:29]: I think that we don't need the TranslateModule here.

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.sandbox.ts [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+import { AuthSandbox } from '../shared/auth/auth.sandbox';
+
+@Injectable()
+export class AdminSandbox extends Sandbox {
+  constructor(protected appState$: Store<store.State>, public authSandbox: AuthSandbox) {
+    super(appState$);
+    this.registerEvents();
+  }
+
+  private registerEvents(): void {}
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.sandbox.ts:14]: Remove this method if we are not using it.

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.sandbox.ts [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import { Injectable } from '@angular/core';
+import { Store } from '@ngrx/store';
+import { Sandbox } from '../shared/sandbox/base.sandbox';
+import * as store from '../shared/store';
+import { AuthSandbox } from '../shared/auth/auth.sandbox';
+
+@Injectable()
+export class AdminSandbox extends Sandbox {
+  constructor(protected appState$: Store<store.State>, public authSandbox: AuthSandbox) {
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/admin.sandbox.ts:9]: authSandbox should be private

File: src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts [hunk 9]
```diff
@@ -1,12 +1,16 @@
...
-                              .then(m => m.DonatorLayoutModule)
-  }
+  {
+    path: '',
+    loadChildren: () =>
+      import('./shared/containers/layout/donator-layout/donator-layout.module').then((m) => m.DonatorLayoutModule),
+  },
+  {
+    path: 'admin',
+    loadChildren: () => import('./admin/admin.module').then((m) => m.AdminModule),
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts:12]: It should be importing the AdminLayoutModule

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.css [hunk 10]
```diff
@@ -1,18 +1,31 @@
...
+  white-space: normal;
+  text-align: center;
+  word-break: break-all;
+}
+
+.logo {
+  width: 100%;
+  padding-left: 1.4em;
+  color: white;
+  font-size: 30px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/components/nav-menu/nav-menu.component.css:11]: Maybe we should change this to em

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/admin-layout/admin-layout.component.css [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+  display: flex !important;
+  align-items: center;
+}
+
+svg {
+  font-size: 2em;
+}
+
+.section-layout {
+  padding: 24px;
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/admin-layout/admin-layout.component.css:27]: Maybe we should change this to another unit such as em, rm, percentage or view port units. If so, apply it to all the stuff where we are using px.

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/admin-layout/admin-layout.component.html [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+      <ul nz-menu>
+        <li *ngFor="let menu of menus" nz-menu-item>
+          <a class="menu-link" routerLink="{{ menu.url }}">
+            <i nz-icon [nzTheme]="'outline'" nzType="{{ menu.iconType }}"></i>
+            {{ menu.title | translate }}
+          </a>
+        </li>
+      </ul>
+    </nz-sider>
+    <nz-layout>
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/admin-layout/admin-layout.component.html:14]: Should we use this layout? Maybe we can use just the section, or remove the section and just use the nz-layout. I dont know for sure, but we could try it.

File: src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts [hunk 13]
```diff
@@ -1,12 +1,16 @@
...
-                            import('./shared/containers/layout/donator-layout/donator-layout.module')
-                              .then(m => m.DonatorLayoutModule)
-  }
+  {
+    path: '',
+    loadChildren: () =>
+      import('./shared/containers/layout/donator-layout/donator-layout.module').then((m) => m.DonatorLayoutModule),
+  },
+  {
+    path: 'admin',
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts:11]: Should the admin module use the auth.guard? So only authenticated users and with a specific role can access to it. @Faculewis 

Maybe the role part could be done later.

Comment (Faculewis) [reply src/DonateTo.WebApplication/ClientApp/src/app/app-routing.module.ts:11]: Yes, it should use the auth guard

File: src/DonateTo.WebApplication/ClientApp/i18n/en.json [hunk 14]
```diff
@@ -73,14 +70,35 @@
...
   "ProfileActionBar": {
     "Logout": "Logout"
   },
-
   "PageNotFound": {
     "Title": "Ooops, Page Not Found",
     "Subtitle": "Please, return to the previous page",
     "Button": "Go back"
+  },
+  "Admin": {
```
Comment (nicolasgranata) [inline src/DonateTo.WebApplication/ClientApp/i18n/en.json:81]: Could we represent the tree of components in the json? For example if you have the Admin module and inside of it, an user component, we should create a json maybe like this: 

"Admin": {
     "Title": value,
      "User": {
           "Title": value
       }
}

So when we look up the translation, we could do something like: Admin.User.Title

What do you think about it @Faculewis ?

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/donation/donation.component.html [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+  <div class="create-button">
+    <button nz-button nzType="primary" class="ant-btn-primary">
+      {{ 'Admin.Create' | translate }}
+    </button>
+  </div>
+  <nz-divider></nz-divider>
+  <button class="ant-btn">{{ 'Admin.ClearFilter' | translate }}</button>
+  <button class="ant-btn">{{ 'Admin.ClearFilterAndSorted' | translate }}</button>
+</div>
+<nz-table #basicTable [nzData]="listOfData">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/donation/donation.component.html:12]: Rename basicTable table to a representative name for the data in the table

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/organization/organization.component.html [hunk 16]
```diff
@@ -0,0 +1,33 @@
+<div class="table-operations">
+  <h4 class="section-title">{{ 'Organization.Title' | translate }}</h4>
+  <div class="create-button">
+    <button nz-button nzType="primary" class="ant-btn-primary">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/organization/organization.component.html:4]: Remove class attribute

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/organization/organization.component.html [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+      {{ 'Admin.Create' | translate }}
+    </button>
+  </div>
+  <nz-divider></nz-divider>
+  <button class="ant-btn">{{ 'Admin.ClearFilter' | translate }}</button>
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/organization/organization.component.html:9]: Remove class and add directive nz-button

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/user/user.component.html [hunk 18]
```diff
@@ -0,0 +1,33 @@
+<div class="table-operations">
+  <h4 class="section-title">{{ 'User.Title' | translate }}</h4>
+  <div class="create-button">
+    <button nz-button nzType="primary" class="ant-btn-primary">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/user/user.component.html:4]: Remove class

File: src/DonateTo.WebApplication/ClientApp/src/app/admin/user/user.component.html [hunk 19]
```diff
@@ -0,0 +1,33 @@
...
+      {{ 'User.Associate' | translate }}
+    </button>
+  </div>
+  <nz-divider></nz-divider>
+  <button class="ant-btn">{{ 'Admin.ClearFilter' | translate }}</button>
+  <button class="ant-btn">{{ 'Admin.ClearFilterAndSorted' | translate }}</button>
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/admin/user/user.component.html:10]: Change class for directive nz-button

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts [hunk 20]
```diff
@@ -0,0 +1,47 @@
...
+
+const ICONS: IconDefinition[] = [HeartOutline, TeamOutline, ProfileOutline];
+
+@NgModule({
+  imports: [
+    NzIconModule.forChild(ICONS),
+    AdminLayoutRoutingModule,
+    CommonModule,
+    NzLayoutModule,
+    NzMenuModule,
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.module.ts:30]: Group the modules of nz-zorro-ant all together

File: src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.component.html [hunk 21]
```diff
@@ -0,0 +1,20 @@
+<nz-layout id="main-layout">
+  <app-nav-menu></app-nav-menu>
+  <nz-layout class="admin-layout">
+    <nz-sider nzTheme="light">
```
Comment (Faculewis) [inline src/DonateTo.WebApplication/ClientApp/src/app/shared/containers/layout/admin-layout/admin-layout.component.html:4]: Please, make it responsive

**Pull request merged**